### PR TITLE
CH 9545 : Http Header Improvements

### PR DESCRIPTION
**Pull request by rubikzube**

### CODE REVIEW

File: constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java [hunk 1]
```diff
@@ -88,9 +91,25 @@ public ConstructorIO(String apiToken, String apiKey, boolean isHTTPS, String hos
         } else {
             this.protocol = "http";
         }
+        if (constructorToken != null) {
```
Comment (voran) [inline constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java:94]: I believe the check is redundant. If it's null, you can set it to null which is the same as not setting it.

File: constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java [hunk 2]
```diff
@@ -100,9 +119,9 @@ public ConstructorIO(String apiToken, String apiKey, boolean isHTTPS, String hos
     public boolean verify() throws ConstructorException {
         try {
             HttpUrl url = this.makeUrl("v1/verify");
-            Request request = new Request.Builder()
+            Builder builder = this.makeAuthorizedRequestBuilder();
```
Comment (voran) [inline constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java:122]: Why not inline builder like:
```
Request request = this.makeAuthorizedRequestBuilder()
    .url(url)
    ...
```

File: constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java [hunk 3]
```diff
@@ -574,6 +596,36 @@ protected HttpUrl makeUrl(String path, UserInfo info) throws UnsupportedEncoding
...
+    /**
+     * Creates a builder for an end user request
+     * 
+     * @param info user information if available
+     * @return Request Builder
+     */
+    protected Builder makeUserRequestBuilder(UserInfo info) {
+      Builder builder = new Request.Builder();
+      if (info != null) {
+        if (this.constructorToken != null) {
```
Comment (voran) [inline constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java:619]: With this code `x-cnstrc-token` won't be set even if info is null. Is this intentional?

Comment (rubikzube) [reply constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java:619]: Well played

File: constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java (hunk 2 above)
Comment (rubikzube) [reply constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java:122]: Will change

File: constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java (hunk 1 above)
Comment (rubikzube) [reply constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java:94]: Ah.  yay.

**Pull request merged**